tutoraspects/asset_command_helpers.py

```diff
@@ -537,42 +174,102 @@ def _get_used_chart_dataset_uuids():
+                for k in ignored_uuids.get("datasets"):
+                    all_dataset_uuids.pop(k, None)
+
+            if ignored_uuids and ignored_uuids.get("charts"):
+                for k in ignored_uuids.get("charts"):
+                    all_chart_uuids.pop(k, None)
 
+    if not all_dataset_uuids and not all_chart_uuids:
+        echo(f"{len(all_chart_uuids) + len(all_dataset_uuids)} orphans detected.")
+
+    return all_dataset_uuids, all_chart_uuids
+
+
+def check_orphan_assets(echo):
+    """
+    Warn about any potentially unused assets.
+    """
+    echo("Looking for potentially orphaned assets...")
     echo()
 
+    all_dataset_uuids, all_chart_uuids = _find_orphan_assets(echo)
+
     if all_dataset_uuids:
         echo(click.style("Potentially unused datasets detected:", fg="yellow"))
-        echo("\n".join(sorted(all_dataset_uuids.values())))
+        echo("Add the UUIDs to aspects_asset_list.yaml to be deleted")
+        for x, y in all_dataset_uuids.items():
+            echo(f'{y.get("name")} (UUID: {x})')
 
     if all_chart_uuids:
         echo(click.style("Potentially unused charts detected:", fg="yellow"))
-        echo("\n".join(sorted(all_chart_uuids.values())))
+        echo("Add the UUIDs to aspects_asset_list.yaml to be deleted")
+        for x, y in all_chart_uuids.items():
+            echo(f'{y.get("name")} (UUID: {x})')
 
-    if not all_dataset_uuids and not all_chart_uuids:
-        echo(f"{len(all_chart_uuids) + len(all_dataset_uuids)} orphans detected.")
+
+def delete_aspects_orphan_assets(echo):
+    """
+    Delete any unused charts and datasets whose UUIDs are listed in
+    aspects_assets_list.yaml - these are owned by Aspects and can safely
+    be deleted.
+    """
+    unused_dataset_uuids, unused_chart_uuids = _find_orphan_assets(echo)
+
+    with open(
+        os.path.join(PLUGIN_PATH, "aspects_asset_list.yaml"), "r", encoding="utf-8"
+    ) as file:
+        aspects_assets = yaml.safe_load_all(file)
+
+        delete_count = 0
+        for line in aspects_assets:
+            orphaned_uuids = line.get("orphaned_uuids")
+            for uuid, data in unused_dataset_uuids.items():
+                if orphaned_uuids and orphaned_uuids.get("datasets"):
+                    if uuid in orphaned_uuids.get("datasets"):
+                        echo(
+                            f"Deleting orphan dataset {data.get('name')} (UUID: {uuid})"
+                        )
+                        os.remove(data.get("file_path"))
+                        delete_count += 1
+
+            for uuid, data in unused_chart_uuids.items():
+                if orphaned_uuids and orphaned_uuids.get("charts"):
+                    if uuid in orphaned_uuids.get("charts"):
+                        echo(f"Deleting orphan chart {data.get('name')} (UUID: {uuid})")
+                        os.remove(data.get("file_path"))
+                        delete_count += 1
+
+    echo(f"Deleted {delete_count} assets")
```

Comment:
This would delete the files from the repository code but if an installation was running and old version with outdated assets those still would be created, you need to add a command for clean up as part of the initialization process